packages/wallets/src/features/cashier/screens/WithdrawalErrorScreen/WithdrawalErrorContent.tsx

```diff
@@ -0,0 +1,71 @@
+import { ComponentProps } from 'react';
+import { TSocketError } from '@deriv/api-v2/types';
+import { WalletButton } from '../../../../components';
+import { CryptoWithdrawalErrorCodes } from '../../../../constants/errorCodes';
+
+type TWithdrawalErrorContentProps = {
+    currency?: string;
+    error: TSocketError<'cashier'>['error'];
+    resetError: VoidFunction;
+    setResendEmail: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const getWithdrawalErrorContent = ({ currency, error, resetError, setResendEmail }: TWithdrawalErrorContentProps) => {
```

Comment:
I'm not keen on introducing this pattern. Pretty hard to determine if thats component or react provider or util function, and it returns a mixed bag of goods - too much responsibility. It have all the component logic minus the render function, so it could as well have it and just be a component. 

One option would be to have a "standard" component + couple util functions to help encapsulate individual responsibilities, e.g. "getErrorMessage(error.code)", "getErrorTitle(error.code)". Thats then easy to write tests, easy to update, easy to reuse the same logic and easy to read.

Reply:
Updated the structure again to a more simple readable pattern while keeping the "standard" component approach, please rereview 🙏